jujugui/static/gui/src/app/store/env/bakery.js

```
@@ -0,0 +1,339 @@
+            path, headers, null, null, false, null,
+            this._requestHandler.bind(
+              this, successCallback, failureCallback
+            )
+          );
+      },
+
+      /**
+       Handles the request response from the _makeRequest method, calling the
+       supplied failure callback if the response status was >= 400 or passing
+       the response object to the supplied success callback.
+
+       @method _requestHandler
+       @param {Function} successCallback Called when the api request completes
+       successfully.
+       @param {Function} failureCallback Called when the api request fails
+       with a response of >= 400.
+       @param {Object} response The XHR response object.
+       */
+      _requestHandler: function (successCallback, failureCallback, response) {
+        var target = response.target;
+        if (target.status >= 400) {
+          failureCallback(response);
+          return;
+        }
+        successCallback(response);
+      },
+
+      /**
+       Authenticate  by discharging the macaroon and
+       then set the cookie by calling the authCookiePath provided
+
+       @method authenticate
+       @param {Macaroon} The macaroon to be discharged
+       @param {String} The path where to send put request to set the cookie back
+       @param {Function} The request to be sent again in case of
+       successful authentication
+       @param {Function} The callback failure in case of wrong authentication
+       */
+      _authenticate: function (m, authCookiePath,
+                               requestFunction, failureCallback) {
+        try {
+          macaroon.discharge(
+            macaroon.import(m),
+            this._obtainThirdPartyDischarge.bind(this),
+            this._processDischarges.bind(
+              this,
+              authCookiePath,
+              requestFunction,
+              failureCallback
+            ),
+            failureCallback
+          );
+        } catch (ex) {
```

what exceptions are we expecting to get here?
(or is it standard behaviour to catch all exceptions?)